ext/microprofile-rest-client/src/main/java/org/glassfish/jersey/microprofile/rest/client/OpenTracingInstrumentation.java

```diff
@@ -0,0 +1,321 @@
+
+            // If there isn't a config override for the method, check if the method has the annotation
+            if (!tracedOverride.isPresent()) {
+                logger.log(Level.FINER, "No config override for annotated method, checking if the method is "
+                        + "annotated directly...");
+                // If the method is annotated directly, simply use the value from it and move on
+                if (method.getAnnotation(Traced.class) != null) {
+                    logger.log(Level.FINER, "Traced annotation found on method.");
+                    // If the method is annotated, check if it's explicitly been set to skip tracing
+                    if (!method.getAnnotation(Traced.class).value()) {
+                        // If we should skip tracing on this method, try to get the path annotation from the method
+                        // after checking if the path annotation has any overrides
+                        jaxrsMethodEndpoint = getMethodLevelPathValueIfPresentUsingConfig(config, method,
+                                annotatedClassCanonicalName, annotatedMethodName, jaxrsMethodEndpoint);
+
+                        shouldSkipTracing = true;
+                    }
+                } else {
+                    // If there wasn't a config override for the method and the method wasn't annotated directly,
+                    // check if there's one for the class
+                    logger.log(Level.FINER, "No config override for annotated method, getting config override "
+                            + "for the annotated class...");
+                    tracedOverride = config.getOptionalValue(annotatedClassCanonicalName + "/"
+                            + Traced.class.getSimpleName() + "/" + "value", boolean.class);
+
+                    // If we found a config override, and it disables the method, add it to the list
+                    if (tracedOverride.isPresent()) {
+                        logger.log(Level.FINER, "Config override found on the annotated class.");
+                        if (!tracedOverride.get()) {
+                            // Since the method doesn't have anything to override this class-level setting, add
+                            // it to the skip list after determining if it has its own path
+                            jaxrsMethodEndpoint = getMethodLevelPathValueIfPresentUsingConfig(config, method,
+                                    annotatedClassCanonicalName, annotatedMethodName, jaxrsMethodEndpoint);
+
+                            shouldSkipTracing = true;
+                        }
+                    } else {
+                        // If we didn't find a config override, check if the class is annotated and use the value
+                        // from that if it is
+                        Traced tracedAnnotation = restClientInterface.getAnnotation(Traced.class);
```

Comment:
This is very similar code to the annotation reader in the main codebase. Is there anything that can be combined?

Reply:
Not easily - I can't depend on stuff in Jersey from Payara, and some of the stuff in Payara depends on other stuff in Payara.